GAE/src/org/waterforpeople/mapping/app/web/DataProcessorRestServlet.java

```diff
@@ -1504,5 +1510,59 @@ private void scheduleCascadeNodeDeletion(Long cascadeResourceId, Long parentNode
                             cascadeResourceId, parentNodeId), e);
         }
     }
+    
+    private void assembleDatapointName(Long surveyId, Long surveyedLocaleId) {
+        final SurveyedLocaleDao slDao = new SurveyedLocaleDao();
+        final SurveyInstanceDAO siDao = new SurveyInstanceDAO();
+        final QuestionDao qDao = new QuestionDao();
+        final QuestionAnswerStoreDao qasDao = new QuestionAnswerStoreDao();
+        
+        List<SurveyedLocale> locales = null;
+        if (surveyedLocaleId != null) {
+            // Single locale update
+            locales = new ArrayList<>();
+            locales.add(slDao.getById(surveyedLocaleId));
+        } else {
+            // get locales by createdSurveyId
+            locales = slDao.listLocalesByCreationSurvey(surveyId, Constants.ALL_RESULTS, null);
+        }
+
+        List<Question> nameQuestions = qDao.listDisplayNameQuestionsBySurveyId(surveyId);
+        
+        SurveyInstance si;
```

Comment:
The variable is initialized each iteration so yes, it's probably more readable :+1: